integration-tests/contract-execute-on-chain-view.spec.ts

```diff
@@ -1,19 +1,20 @@
 import { codeViewsTopLevel } from "../packages/taquito-local-forging/test/data/contract_views_top_level";
 import { CONFIGS } from "./config";
 import BigNumber from 'bignumber.js';
-import { Protocols } from "@taquito/taquito";
-import { HttpResponseError } from "@taquito/http-utils";
+import { Protocols, ViewSimulationError } from "@taquito/taquito";
 
-CONFIGS().forEach(({ lib, rpc, setup }) => {
+CONFIGS().forEach(({ lib, protocol, rpc, setup }) => {
   const Tezos = lib;
- 
+  const jakartanet = protocol === Protocols.PtJakart2 ? test: test.skip;
+  const kathmandunet = protocol === Protocols.PtKathman ? test: test.skip;
+
   describe(`On chain views using the contract API: ${rpc}`, () => {
 
     beforeEach(async (done) => {
       await setup(true)
       done()
     })
-    test(`As a user I want to originate a smart contract having top level views and simulate the views execution`, async (done) => {
+    kathmandunet(`As a user I want to originate a smart contract having top level views and simulate the views execution`, async (done) => {
```

Comment:
Before the transaction to Kathmandu protocol, we need to stay compatible with Jakarta.
We should not skip the test on jakartanet. That said the contractViews would need to use the old implementation if the protocol is jakarta and the new one if the protocol is Kathmandu.